packages/core/test/lib/tracing/trace.test.ts

```diff
@@ -333,6 +333,121 @@ describe('startSpan', () => {
     expect(getActiveSpan()).toBe(undefined);
   });
 
+  it('allows to force a transaction with isTransaction=true', async () => {
+    const options = getDefaultTestClientOptions({ tracesSampleRate: 1.0 });
+    client = new TestClient(options);
+    setCurrentClient(client);
+    client.init();
+
+    const transactionEvents: Event[] = [];
+
+    client.addEventProcessor(event => {
+      if (event.type === 'transaction') {
+        transactionEvents.push(event);
+      }
+      return event;
+    });
+
+    startSpan({ name: 'outer transaction' }, () => {
+      startSpan({ name: 'inner span' }, () => {
+        startSpan({ name: 'inner transaction', isTransaction: true }, () => {
+          startSpan({ name: 'inner span 2' }, () => {
+            // all good
+          });
+        });
+      });
+    });
+
+    await client.flush();
+
+    const normalizedTransactionEvents = transactionEvents.map(event => {
+      return {
+        ...event,
+        // eslint-disable-next-line deprecation/deprecation
+        spans: event.spans?.map(span => ({ name: span.name, id: span.spanContext().spanId })),
+        sdkProcessingMetadata: {
+          dynamicSamplingContext: event.sdkProcessingMetadata?.dynamicSamplingContext,
+        },
+      };
+    });
+
+    expect(normalizedTransactionEvents).toHaveLength(2);
+
+    const outerTransaction = normalizedTransactionEvents.find(event => event.transaction === 'outer transaction');
+    const innerTransaction = normalizedTransactionEvents.find(event => event.transaction === 'inner transaction');
+
+    const outerTraceId = outerTransaction?.contexts?.trace?.trace_id;
+    // The inner transaction should be a child of the last span of the outer transaction
+    const innerParentSpanId = outerTransaction?.spans?.[0].id;
+    const innerSpanId = innerTransaction?.contexts?.trace?.span_id;
+
+    expect(outerTraceId).toBeDefined();
+    expect(innerParentSpanId).toBeDefined();
+    expect(innerSpanId).toBeDefined();
+    // inner span ID should _not_ be the parent span ID, but the id of the new span
+    expect(innerSpanId).not.toEqual(innerParentSpanId);
+
+    expect(outerTransaction).toEqual(
```

Comment:
can we split these asserts up a bit? It's always a huge pain to debug what is mismatching as soon as something fails.